java/src/server/plugin/PluginLoader.java

```diff
@@ -1,27 +1,94 @@
 package server.plugin;
-
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
-public class PluginLoader implements IPluginLoader {
+import static java.util.Arrays.stream;
 
+public class PluginLoader implements IPluginLoader {
 
+    /**
+     * @inheritDoc
+     */
     @Override
-    public List<IPlugin> loadConfig(List<PluginConfig> configs) {
-        return null;
+    public List<IPlugin> loadConfig(List<PluginConfig> configs, File folder) {
+        List<String> neededPlugins = configs.stream()
+                .map(o -> o.getName().toLowerCase())
+                .collect(Collectors.toList());
+
+        return Arrays.stream(folder.listFiles())
+                .filter(x -> ".jar".equals(x.getName().substring(x.getName().length() - 4, x.getName().length())))
```

Comment:
What I mean is this line with throw an exception if there's a file in the plugin directory named `abc`. `x.getName().length()` would be 3, and subtracted by 4, it would be -1. [`substring` throws an `IndexOutOfBoundsException` if the `beginIndex` is negative.](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#substring-int-int-)The precondition for this filter working properly is that each filename in the directory is at least 4 characters.